server/game/cards/plots/04/battleoftheblackwater.js

```diff
@@ -0,0 +1,29 @@
+const _ = require('underscore');
+
+const PlotCard = require('../../../plotcard.js');
+
+class BattleOfTheBlackwater extends PlotCard {
+    setupCardAbilities() {
+        this.whenRevealed({
+            handler: () => {
+                _.each(this.game.getPlayersInFirstPlayerOrder(), player => {
+                    this.removeAllDupes(player);
+                });
+            }
+        });
+    }
+
+    removeAllDupes(player) {
+        var characters = player.cardsInPlay.filter(card => card.dupes.size() > 0);
+        _.each(characters, character => {
+            while(character.dupes.size() > 0) {
+                var dupe = character.removeDuplicate();
+                player.moveCard(dupe, 'discard pile');
```

Comment:
You should use `player.removeDuplicate(character)` instead, which will do the removal + discard for you and fire the proper event.